src/lava/lib/dl/netx/blocks/process.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier:  BSD-3-Clause
+
+"""Composable blocks processes in Lava."""
+
+from typing import Type, Union
+import numpy as np
+import h5py
+from lava.magma.core.process.ports.ports import InPort, OutPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.proc.dense.process import Dense as DenseSynapse
+from lava.proc.conv.process import Conv as ConvSynapse
+
+
+class AbstractBlock(AbstractProcess):
+    """Abstract block definition.
+
+    Parameters
+    ----------
+    shape : tuple or list
+        shape of the block output in (x, y, z) or WHC format.
+    neuron_params : dict
+        dictionary of neuron parameters. Defaults to None.
+    has_graded_input : dict
+        flag for graded spikes at input. Defaults to False.
+    """
+    def __init__(self, **kwargs: Union[dict, tuple, list, int, bool]) -> None:
+        super().__init__(**kwargs)
+        self.shape: tuple = kwargs.pop('shape')
+        self.neuron_params = kwargs.pop('neuron_params', None)
+        self.neuron_process = None
+        if self.neuron_params is not None:
+            self.neuron_process = self.neuron_params.pop('neuron_proc', None)
+        self.neuron = None
+        self.has_graded_input = kwargs.pop('has_graded_input', False)
+        self.has_graded_output = False
+        if 'use_graded_spikes' in self.neuron_params.keys():
+            self.has_graded_output = self.neuron_params['use_graded_spikes']
+
+    def _clean(self) -> None:
+        del self.neuron_params
+        del self.neuron_process
+
+    def _neuron(self, bias: np.ndarray) -> Type[AbstractProcess]:
+        if bias is None:
+            return self.neuron_process(shape=self.shape, **self.neuron_params)
+        else:
+            if len(self.shape) == 1:  # scalar bias value
+                bias_mant = bias
+            elif list(bias.shape) == list(self.shape):
+                # neuron wise bias value
+                bias_mant = bias
+            elif self.shape[-1] == bias.shape[0]:  # channel wise bias value
+                bias_mant = np.zeros(self.shape)
+                bias_mant[:, :] = bias
+            else:
+                raise RuntimeError(
+                    f'Bias of shape {bias.shape} could not be broadcast to '
+                    f'neuron of shape {self.shape}.'
+                )
+            # print(f'{bias=}')
+            # print(f'{bias_mant=}')
```

Comment:
Please remove or add as debug logging so they can be left out by default but turned on when debugging (`self.log` should now be a part of `AbstractProcess` and inherited here):
```
self.log.debug(f'{bias=}')
self.log.debug(f'{bias_mant=}')
```